pkg/api/validation/schema.go

```diff
@@ -219,6 +254,40 @@ func (s *SwaggerSchema) ValidateObject(obj interface{}, fieldName, typeName stri
 	return allErrs
 }
 
+// delegateForApiVersion delegates the validation of an object in a different ApiGroup to another SwaggerSchema.
+// Will panic if obj is in the same api group as this SwaggerSchema.
+func (s *SwaggerSchema) delegateForApiVersion(obj interface{}) error {
+	// Sanity check to prevent infinite recursion from delegating to ourselves
+	if !s.isDifferentApiVersion(obj) {
+		panic(fmt.Errorf("Cannot delegate schema for object %+v in same api version: %s", obj, s.api.ApiVersion))
```

Comment:
This represents a coding error since isDifferentApiVersion must be called before this.  This is not something we can recover from.  I will clarify in the comment.